assets/js/admin/first-course-creation-notice.js

```diff
@@ -0,0 +1,181 @@
+export const hasLessonInOutline = ( blocks ) => {
+	return blocks.some( ( block ) => {
+		if ( block.name === 'sensei-lms/course-outline-lesson' ) {
+			return true;
+		}
+
+		if ( block.innerBlocks?.length ) {
+			return hasLessonInOutline( block.innerBlocks );
+		}
+
+		return false;
+	} );
+};
+
+// Detect when a user selects a Sensei pattern.
+let selectedSenseiPattern = false;
+
+export const isSenseiPatternSelected = () => selectedSenseiPattern;
+
+const rewrittenActions = {};
+const originalActions = {};
+
+const checkIfAnySenseiPatternGotInserted = ( ...args ) => {
+	const meta = findObject( args, ( item ) => item?.patternName );
+	let patternName = meta?.patternName;
+	if ( patternName && patternName.includes( 'sensei' ) ) {
+		selectedSenseiPattern = true;
+	}
+};
+
+const REDUX_ACTION_OVERRIDERS = {
+	'core/block-editor': {
+		insertBlocks: checkIfAnySenseiPatternGotInserted,
+	},
+	'core/editor': {
+		resetEditorBlocks: checkIfAnySenseiPatternGotInserted,
+	},
+};
+
+use( ( registry ) => ( {
+	dispatch: ( namespace ) => {
+		const namespaceName =
+			typeof namespace === 'object' ? namespace.name : namespace;
+		const actions = { ...registry.dispatch( namespaceName ) };
+		const overriderActions = REDUX_ACTION_OVERRIDERS[ namespaceName ];
+
+		if ( overriderActions ) {
+			Object.keys( overriderActions ).forEach( ( actionName ) => {
+				const originalAction = actions[ actionName ];
+				const newAction = overriderActions[ actionName ];
+
+				if ( ! rewrittenActions[ namespaceName ] ) {
+					rewrittenActions[ namespaceName ] = {};
+				}
+				if ( ! originalActions[ namespaceName ] ) {
+					originalActions[ namespaceName ] = {};
+				}
+
+				if (
+					! originalActions[ namespaceName ][ actionName ] ||
+					originalActions[ namespaceName ][ actionName ] !==
+						originalAction
+				) {
+					originalActions[ namespaceName ][
+						actionName
+					] = originalAction;
+					rewrittenActions[ namespaceName ][ actionName ] = (
+						...args
+					) => {
+						try {
+							newAction( ...args );
+						} catch ( err ) {}
+						return originalAction( ...args );
+					};
+				}
+
+				actions[ actionName ] =
+					rewrittenActions[ namespaceName ][ actionName ];
+			} );
+		}
+
+		return actions;
+	},
+} ) );
+
+export const handleFirstCourseCreationHelperNotice = () => {
+	const { createInfoNotice, removeNotice } = dispatch( 'core/notices' );
+	const userId = select( 'core' ).getCurrentUser()?.id;
+	const firstCourseNoticeDismissedKey =
+		'sensei-lms-first-course-notice-dismissed-' + userId;
+	const isFirstCourseNoticeDismissed = !! window.localStorage.getItem(
+		firstCourseNoticeDismissedKey
+	);
+	const noticeId = 'course-outline-block-setup-incomplete';
+	const isNewCourse = window?.sensei?.isNewCourse;
+
+	let noticeCreated = false;
+	let noticeRemoved = false;
+
+	const notice = __(
+		'Nice! Now you can <a href="javascript:;" onclick="window?.handleCourseOutlineBlockIncomplete();">add some lessons</a> to your course.',
```

Comment:
If we're good with the design change, we can go for it 👍 Our current design of showing it as a URL looks better to me. But as showing it as button is simpler, as that's the default behavior. I think the main benefit of this will be not having to use the `__unstableHTML` property.

Reply:
I'm fine with changing it to use a button if it helps simplify things. In that case, we could change the message to:

> Nice! Now you can add some content to your course. Add Lessons

where _Add Lessons_ is a button that jumps to / adds the Course Outline block.

Let's see what that looks like. 🙂

Reply:
Updated comment [here](https://github.com/Automattic/sensei/pull/7364#issuecomment-1881521558) 🙂